Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -452,127 +435,142 @@ void AODProducerWorkflowDPL::addToFwdTracksTable(FwdTracksCursorType& fwdTracksC
+    fwdInfo.trackTimeRes = o2::constants::lhc::LHCBunchSpacingNS * bcWidth / 2; // half interval
     errGaussian = false;
 
   } else {
     // This is a GlobalMuonTrack or a GlobalForwardTrack
     const auto track = data.getGlobalFwdTrack(trackID);
-    x = track.getX();
-    y = track.getY();
-    z = track.getZ();
-    phi = track.getPhi();
-    tanl = track.getTanl();
-    invqpt = track.getInvQPt();
-    chi2 = track.getTrackChi2();
-    // nClusters = track.getNumberOfPoints();
-    chi2matchmchmid = track.getMIDMatchingChi2();
-    chi2matchmchmft = track.getMFTMCHMatchingChi2();
-    matchscoremchmft = track.getMFTMCHMatchingScore();
-    matchmfttrackid = track.getMFTTrackID();
-    matchmchtrackid = track.getMCHTrackID();
-    trackTime = track.getTimeMUS().getTimeStamp() * 1.e3;
-    trackTimeRes = track.getTimeMUS().getTimeStampError() * 1.e3;
-
-    sigX = TMath::Sqrt(track.getCovariances()(0, 0));
-    sigY = TMath::Sqrt(track.getCovariances()(1, 1));
-    sigPhi = TMath::Sqrt(track.getCovariances()(2, 2));
-    sigTgl = TMath::Sqrt(track.getCovariances()(3, 3));
-    sig1Pt = TMath::Sqrt(track.getCovariances()(4, 4));
-    rhoXY = (Char_t)(128. * track.getCovariances()(0, 1) / (sigX * sigY));
-    rhoPhiX = (Char_t)(128. * track.getCovariances()(0, 2) / (sigPhi * sigX));
-    rhoPhiY = (Char_t)(128. * track.getCovariances()(1, 2) / (sigPhi * sigY));
-    rhoTglX = (Char_t)(128. * track.getCovariances()(0, 3) / (sigTgl * sigX));
-    rhoTglY = (Char_t)(128. * track.getCovariances()(1, 3) / (sigTgl * sigY));
-    rhoTglPhi = (Char_t)(128. * track.getCovariances()(2, 3) / (sigTgl * sigPhi));
-    rho1PtX = (Char_t)(128. * track.getCovariances()(0, 4) / (sig1Pt * sigX));
-    rho1PtY = (Char_t)(128. * track.getCovariances()(1, 4) / (sig1Pt * sigY));
-    rho1PtPhi = (Char_t)(128. * track.getCovariances()(2, 4) / (sig1Pt * sigPhi));
-    rho1PtTgl = (Char_t)(128. * track.getCovariances()(3, 4) / (sig1Pt * sigTgl));
-
-    trackTypeId = (chi2matchmchmid >= 0) ? o2::aod::fwdtrack::GlobalMuonTrack : o2::aod::fwdtrack::GlobalForwardTrack;
+    fwdInfo.x = track.getX();
+    fwdInfo.y = track.getY();
+    fwdInfo.z = track.getZ();
+    fwdInfo.phi = track.getPhi();
+    fwdInfo.tanl = track.getTanl();
+    fwdInfo.invqpt = track.getInvQPt();
+    fwdInfo.chi2 = track.getTrackChi2();
+    // fwdInfo.nClusters = track.getNumberOfPoints();
+    fwdInfo.chi2matchmchmid = track.getMIDMatchingChi2();
+    fwdInfo.chi2matchmchmft = track.getMFTMCHMatchingChi2();
+    fwdInfo.matchscoremchmft = track.getMFTMCHMatchingScore();
+    fwdInfo.matchmfttrackid = track.getMFTTrackID();
+    fwdInfo.matchmchtrackid = track.getMCHTrackID();
+    fwdInfo.trackTime = track.getTimeMUS().getTimeStamp() * 1.e3;
+    fwdInfo.trackTimeRes = track.getTimeMUS().getTimeStampError() * 1.e3;
+
+    getMCHBitMap(fwdInfo.matchmchtrackid);
+
+    int midTrackID = track.getMIDTrackID();
```

Comment:
The information that you want to put in the AOD is now computed in the MID Track. 
So it's enough to do: midTrack.midBitMap = midTrack.getHitMap().
Concerning the second information, this was changed with respect to the midBoards information in a way that it is more similar to what was done in Run 2.
If you prefer, I can add a getEfficiencyWord method that you can use to define midBoards.

Reply:
Hi @dstocco. Thank you for the PR #8495. I have modified the code using the method you added. Could you please take a look?

Reply:
I think it should be ok now. Thanks!